src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -360,20 +367,37 @@ def expand_list(old):
     return new
 
 
-def find_files(path, pattern='*', kind=''):
+def find_files(path, pattern='*', kind='',
+               follow_links=False, follow_mounts=True):
     """
     Return a list of files similar to the find command
     """
     if isinstance(pattern, str):
         pattern = [pattern]
     if isinstance(kind, str):
-        kind = [kind]
+        if not kind:
+            kind = []
+        else:
+            kind = [kind]
     if not isinstance(pattern, list):
-        raise TypeError('Pattern must be a string or list.')
+        raise TypeError('Pattern must be a string or list')
     if not isinstance(kind, list):
-        raise TypeError('Kind must be a string or list.')
-    for root, dirs, files in os.walk(path):
+        raise TypeError('Kind must be a string or list')
+    # Top level not excluded if it is a mount point
+    mounts = []
+    for root, dirs, files in os.walk(path, followlinks=follow_links):
         for name in [os.path.join(root, x) for x in dirs + files]:
+            if not follow_mounts:
+                if os.path.isdir(name) and os.path.ismount(name):
+                    mounts.append(os.path.join(name, ''))
+                    continue
+                undermount = False
+                for mountpoint in mounts:
+                    if name.startswith(mountpoint):
+                        undermount = True
+                        break
+                if undermount:
+                    continue
```

Comment:
The os.walk() method takes a parameter called followlinks to determine whether symbolic links are to be avoided. It does not offer a similar parameter (like followmounts) to avoid mount points it encounters. We maintain a list of mount points encountered in the mounts variable. When we encounter a mount, it gets add it to the list. As the search progresses we check to see if the current item is under a mount point. If it is, we skip it when follow_mounts is false.